### PR TITLE
GrafanaUI: Improved ClipboardButton success state

### PR DESCRIPTION
**Pull request by joshhunt**
**What this PR does / why we need it**:

Iterates on @evictorero's PR https://github.com/grafana/grafana/pull/52069 to add an inline mini-toast (crustini?) above the button when copied.

Changes:

 - Add new InlineToast component that, similar to a tooltip, attaches to another element with a little animation. The idea is we would use this for future similar use cases (such as a "saved" indicator next to an auto-saved form control?)
 - Uses Popper + InlineToast in ClipboardButton to show success
 - Does some funky stuff to overlay just a check in the ClipboardButton during success state

Indicator:

image

ClipboardButton success state:

image

https://user-images.githubusercontent.com/46142/180469040-79171ce9-31e7-40ac-879c-733f145df3b7.mp4

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

A few pending questions would like resolved:

 - [x] Name for the `<Indicator />` component.  We don't think Pill is appropriate, as it's not supposed to be a generic pill-shaped control, but rather one specific for denoting statuses. Other candidates we had were Feedback or InlineToast.
    - Renamed to InlineToast
 - [x] Should ~Indicator~ InlineToast have the animation itself? Should the animation be abstracted out into another component?
    - The component now has the Portal/positioning/animation all in itself.
 - [x] The button overlay thing - to show _just_ a check in the button without affecting the size of the button - is a big kludgy hack. Is there a better way to do this? I was hesitant to put into into Button itself - I don't want it to become too overloaded with functionality. Should this also be in another component (I don't see how with the current implementation)?

### CODE REVIEW

**Comment by staton-hyse11** [comment]
[GLDS Component Requirement Doc](https://docs.google.com/document/d/1L9X4r9yQwP0myrL5OLBIHKrAGhUp8AqY72YByuy6_ts/edit#) for reference on component documentation

File: packages/grafana-ui/src/components/Indicator/Indicator.story.internal.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  );
+};
+
+export const WithAButton: Story<IndicatorProps> = (args) => {
+  return (
+    <ClipboardButton icon="copy" getText={() => 'hello world'}>
+      Copy surprise
+    </ClipboardButton>
+  );
+};
```
Comment (leventebalogh) [inline packages/grafana-ui/src/components/Indicator/Indicator.story.internal.tsx:41]: **Question:** what is the reason for not exposing this component?

Comment (joshhunt) [reply packages/grafana-ui/src/components/Indicator/Indicator.story.internal.tsx:41]: mistake! didn't notice!

File: packages/grafana-ui/src/components/ClipboardButton/ClipboardButton.tsx [hunk 2]
```diff
@@ -83,3 +105,24 @@ const copyText = async (text: string, buttonRef: React.MutableRefObject<HTMLButt
...
+      }),
+    }),
+    successOverlay: css({
+      position: 'absolute',
+      top: 0,
+      bottom: 0,
+      right: 0,
+      left: 0,
+      visibility: 'visible', // re-visible the overlay
+    }),
```
Comment (joshhunt) [inline packages/grafana-ui/src/components/ClipboardButton/ClipboardButton.tsx:126]: I want to change the button to show _just_ a check, but not change the size. This was the best I could come up with, but it's pretty icky.

I was hesitant to add this functionality directly into Button

Comment (JoaoSilvaGrafana) [reply packages/grafana-ui/src/components/ClipboardButton/ClipboardButton.tsx:126]: The only other way I can think of is if you check the size of the Button after first render and then just keep it to that with any change, but I think that would be even _more_ hacky than doing an overlay...

**Review by JoaoSilvaGrafana — APPROVED** [review]

**Pull request merged**